crates/typed-store/src/rocks/iter.rs

```diff
@@ -32,15 +42,28 @@ impl<'a, K: DeserializeOwned, V: DeserializeOwned> Iterator for Iter<'a, K, V> {
 
     fn next(&mut self) -> Option<Self::Item> {
         if self.db_iter.valid() {
+            let _timer = self
+                .db_metrics
+                .rocksdb_iter_latency_seconds
+                .with_label_values(&[&self.cf])
+                .start_timer();
             let config = bincode::DefaultOptions::new()
                 .with_big_endian()
                 .with_fixint_encoding();
-            let key = self.db_iter.key().and_then(|k| config.deserialize(k).ok());
-            let value = self
+            let raw_key = self
+                .db_iter
+                .key()
+                .expect("Valid iterator failed to get key");
+            let raw_value = self
                 .db_iter
                 .value()
-                .and_then(|v| bincode::deserialize(v).ok());
-
+                .expect("Valid iterator failed to get value");
+            self.db_metrics
```

Comment:
I wonder how expensive it would be to monitor every write.  I think this is valuable, although write byte size can be obtained after the fact by going through a database offline.

Reply:
I think a value we are getting here is the histogram of the byte sizes in real time which may be valuable in catching some tail latency kind of issues. I don't know if we should worry about perf for updating metrics in this path (there is so much else we are doing here i.e. serialization/deserialization that updating an in memory counter would dwarf in comparison